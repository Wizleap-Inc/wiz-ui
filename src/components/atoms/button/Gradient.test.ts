import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import GradientButton from "./Gradient.vue";

describe("GradientButton", () => {
  describe("Mount", () => {
    it("正常にレンダリングされるか", () => {
      const wrapper = mount(GradientButton);
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe("Slot", () => {
    it("テキストが正しく表示されるか", () => {
      const slotText = "Hello World!";
      const wrapper = mount(GradientButton, {
        slots: {
          default: slotText,
        },
      });
      expect(wrapper.html()).toContain(slotText);
    });
  });

  describe("Props", () => {
    it("disabledが正しく設定されるか", () => {
      const wrapper = mount(GradientButton, {
        propsData: {
          disabled: true,
        },
      });
      expect(wrapper.classes()).toContain("gradient-btn--disabled");
    });
  });

  describe("Emits", () => {
    it("clickイベントが発火されるか", async () => {
      const wrapper = mount(GradientButton);
      const button = wrapper.find("button");
      await button.trigger("click");
      expect(button.emitted("click")).toBeTruthy();
    });

    it("disabled時clickイベントが発火されないか", async () => {
      const wrapper = mount(GradientButton, {
        propsData: {
          disabled: true,
        },
      });
      const button = wrapper.find("button");
      await button.trigger("click");
      expect(button.emitted("click")).toBeFalsy();
    });
  });
});
