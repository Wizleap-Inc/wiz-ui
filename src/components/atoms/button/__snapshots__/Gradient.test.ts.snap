// Vitest Snapshot v1

exports[`GradientButton > Mount > 正常にレンダリングされるか 1`] = `"<button data-v-a3b6d44e=\\"\\" class=\\"gradient-btn\\" style=\\"--a3b6d44e-GRADIENT_COLOR: linear-gradient(90deg, #3db783 0%, #099ec3 100%);\\"></button>"`;
